pkg/runtime/field_export_reconciler.go

```diff
@@ -60,18 +59,22 @@ var (
 // It implements the upstream controller-runtime `Reconciler` interface.
 type fieldExportReconciler struct {
 	reconciler
+	// resourceReconciler indicates that the reconciler should be reconciling
+	// AWS resources
+	resourceReconciler bool
 	// rd is only used if binding to an ACK resource (not `FieldExport`)
-	rd *acktypes.AWSResourceDescriptor
+	rd acktypes.AWSResourceDescriptor
```

Comment:
zero value for the interface is nil. 
I do not think you need the "resourceReconciler" boolean. You can just use nil check for rd

Reply:
When I tested that it wasn't really working. It was being instantiated as `acktypes.AWSResourceDescriptor{}` (the empty object). The nil check was failing because of that. I'm not very familiar with nillable interfaces in Go

Reply:
I just tested by creating a small struct and the value of AWSResourceDescriptor was nil when not initialized in the struct. Can you please double check your implementation again? Maybe try using `ackCompare.IsNil` function.

I used following test to verify the nil check.

```
type mystruct struct {
	mystring string
	resDes   acktypes.AWSResourceDescriptor
}

func TestMyStruct(t *testing.T) {
	myS := mystruct{mystring: "myString"}
	if myS.resDes == nil {
		fmt.Println("resDes is nil")
	} else {
		fmt.Println("resDes is not nil")
	}
	fmt.Println(myS.mystring)
}
```